mmcls/evaluation/metrics/multi_task.py

```diff
@@ -2,24 +2,78 @@
 from typing import Dict, List, Sequence
 
 from mmengine.evaluator import BaseMetric
-from mmengine.registry import METRICS as mmengineMetrics
 
 from mmcls.registry import METRICS
+from mmcls.structures import MultiTaskDataSample
 
 
 @METRICS.register_module()
-class MultiTasks(BaseMetric):
-    """a."""
+class MultiTasksMetric(BaseMetric):
+    """Metrics for MultiTask
+    Args:
+        task_metrics(dict): a dictionary in the keys are the names of the tasks
+            and the values is a list of the metric corresponds to this task
+    Examples:
+        task_metrics = {
+            'task0': [dict(type='Accuracy', topk=(1, ))],
+            'task1': [
+                dict(type='Accuracy', topk=(1, 3)),
+                dict(type='SingleLabelMetric', items=['precision', 'recall'])
+            ]
+        }
+        task_metrics2 = {
+            'task0': [dict(type='Accuracy', topk=(1, ))],
+            'task1': {
+                'task10': [
+                    dict(type='Accuracy', topk=(1, 3)),
+                    dict(type='SingleLabelMetric', items=['precision'])
+                ],
+                'task11': [dict(type='Accuracy', topk=(1, ))]
+            }
+        }
```

Comment:
Please rework the example as in https://github.com/marouaneamz/mmclassification/blob/5d26026f4b1f410df847c787fcabf90cee63473e/mmcls/evaluation/metrics/single_label.py#L75-L101